Repalce `dirname(__FILE__)` with `__DIR__`

Ayesh opened the pull request:
Pull Request for Issue # .

### Summary of Changes

PHP 5.3 adds support for the `__DIR__` magic constant, and its result is identical to `dirname(__FILE__)`. The former is more clearer and does not has a function call, giving a tiny performance advantage as well.

### Testing Instructions



### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request



### Documentation Changes Required

Quy commented:
Please revert restore.php. https://github.com/joomla/joomla-cms/pull/14269#issuecomment-282790995

richard67 commented:
> PHP 5.3 adds support for the `__DIR__` magic constant, and its result is identical to `dirname(__FILE__)`.

Can we be sure that this is true for symbolic links, too?

Ayesh commented:
> 
> 
> > PHP 5.3 adds support for the `__DIR__` magic constant, and its result is identical to `dirname(__FILE__)`.
> 
> Can we be sure that this is true for symbolic links, too?

Hi @richard67 - yes. `__DIR__` will have the same result as if we were to `dirname(__FILE__)`. 

Both `__FILE__` and `__DIR__` resolve symlink, and always use realpaths; not the symlink paths.

zero-24 commented:
Please revert the changes in the restore.php as that is threated as external code. The change in the action logs would than be fine but why do we use that constant there directly anyway?

Did't we use a different constant in other places?

Ayesh committed:
Repalce `dirname(__FILE__)` with `__DIR__`

PHP 5.3 adds support for the `__DIR__` magic constant, and its result is identical to `dirname(__FILE__)`. The former is more clearer and does not has a function call, giving a tiny performance advantage as well.

Ayesh commented:
Hi @zero-24, I force-pushed with changes in restore.php reverted. 
I don't really know the answer to the second question, sorry. I was randomly checking out the code and wanted to send a PR for the odd improvement I came across.

SharkyKZ commented:
I have tested this item :white_check_mark: successfully on 8dafe8e11014046e0d3ea4282660ad01edd08ae5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31299">issues.joomla.org/tracker/joomla-cms/31299</a>.</sub>

Quy commented:
I have tested this item :white_check_mark: successfully on 8dafe8e11014046e0d3ea4282660ad01edd08ae5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31299">issues.joomla.org/tracker/joomla-cms/31299</a>.</sub>

Quy commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31299">issues.joomla.org/tracker/joomla-cms/31299</a>.</sub>

HLeithner commented:
Thanks
